### DISCUSSION TITLE
Event hook plugin for Apache APISIX

### DISCUSSION THREAD

**Discussion by Boburmirzo** [apache/apisix#8592] [Ideas]
It is nice to have a new webhook event handler as a plugin. 

Event hooks are outbound calls from API Gateway. With event hooks, the Apache APISIX API Gateway can communicate with target services or resources, letting the target know that an event was triggered. When an event is triggered in Apisix, it calls a URL with information about that event.  Parts of the plugin codebase can subscribe to these events, then process the events using callbacks. 

Examples from Kong Enterprise and Tyk:

https://docs.konghq.com/gateway/latest/kong-enterprise/event-hooks/
https://tyk.io/docs/basic-config-and-security/report-monitor-trigger-events/webhooks/

**Comment by juzhiyuan** [comment]
Hi @Boburmirzo, this feature is interesting indeed 😄 Usually the applications we're using support WebHook features, but I personally don't have enough scenarios to put this feature on Gateway.

Do you have more information, like what cases we should rely on Gateway to send webhooks?

**Reply by Boburmirzo** [reply]
@juzhiyuan Good! There are many use cases. The simple idea is to make it possible for users to subscribe to events (By registering a unique URL in a new plugin) in Apisix. For example, when authentication fails for consumers, Apisix will send a notification to the client-registered URL by making a Post request.

**Reply by juzhiyuan** [reply]
>  when authentication fails for consumers, Apisix will send a notification to the client-registered URL by making a Post request.

This reminds me a lot 👍👍

**Reply by Boburmirzo** [reply]
Let me know if we get votes from more people, we can work on it.

**Reply by Boburmirzo** [reply]
Can we invite others to this discussion? Or should I notify others by email?

**Reply by juzhiyuan** [reply]
> Can we invite others to this discussion? Or should I notify others by email?

This is the public channel, of course can invite people attend :)
